src/formatDate.js

```diff
@@ -8,7 +8,31 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splittedDate = date.split(fromFormat[fromFormat.length - 1]);
+  const dateProperties = {};
+  const updatedDate = [];
+
+  for (let i = 0; i < 3; i++) {
+    dateProperties[fromFormat[i]] = splittedDate[i];
+  }
```

Comment:
The loop here assumes there will always be three date parts, which is a safe assumption given the task constraints. However, if in the future the function needs to support more complex date formats, this approach might become a limitation.